pkg/runtime/reconciler.go

```diff
@@ -557,20 +543,31 @@ func (r *resourceReconciler) setResourceUnmanaged(
 
 	orig := res.RuntimeObject().DeepCopyObject()
 	r.rd.MarkUnmanaged(res)
-	rlog.Enter("kc.Patch (metadata + spec)")
-	err = r.kc.Patch(
-		ctx,
-		res.RuntimeObject(),
-		client.MergeFrom(orig),
-	)
-	rlog.Exit("kc.Patch (metadata + spec)", err)
+	err = r.patchResourceMetadataAndSpec(ctx, r.rd.ResourceFromRuntimeObject(orig), res)
 	if err != nil {
 		return err
 	}
 	rlog.Debug("removed resource from management")
 	return nil
 }
 
+// failOnResourceUnmanaged ensures that the underlying CR in the supplied
+// AWSResource has a finalizer. If it does not, it will set a Terminal condition
+// and return with an error
+func (r *resourceReconciler) failOnResourceUnmanaged(
+	ctx context.Context,
+	res acktypes.AWSResource,
+) error {
+	if r.rd.IsManaged(res) {
+		return nil
+	}
+
+	notManagedMessage := "Resource already exists"
+	notManagedReason := "This resource already exists and is not managed by ACK"
```

Comment:
Let's provide a little more help to the user. How about this instead?

```go
msg := "This resource already exists but is not managed by ACK. "+
    "To bring the resource under ACK management, you should explicitly adopt "+
    "the resource by creating a services.k8s.aws/AdoptedResource"
```

And since that's pretty long, why not place that text as a constant in the `pkg/conditions` package and use it here.